framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -189,6 +214,160 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		}
 	}
 
+	/**
+	 * @see https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#cross-chain-update-transaction-properties
+	 * This function computes CCU params especially, certificate, activeValidatorsUpdate and inboxUpdate
+	 * - Uses either lastCertificate or newCertificate
+	 * - When lastCertificate, it only computes pending CCMs if any else it skips CCU creation
+	 * - When newCertificate it computes certificate, activeValidatorsUpdate and inboxUpdate
+	 */
+	private async _computeCCUParams(
+		blockHeaders: BlockHeader[],
+		aggregateCommits: AggregateCommit[],
+		validatorsHashPreimage: ValidatorsData[],
+		ccmsFromEvents: CCMsFromEvents[],
```

Comment:
its an internal method so maybe not so important to have interface for the args but surely we can do out side this PR as it will trigger changes in tests